src/java/org/apache/poi/ss/usermodel/DataFormatter.java

```diff
@@ -315,62 +311,72 @@ private Format getFormat(Cell cell, ConditionalFormattingEvaluator cfEvaluator)
 
     private Format getFormat(double cellValue, int formatIndex, String formatStrIn) {
         localeChangedObservable.checkForLocaleChange();
-        
-//      // Might be better to separate out the n p and z formats, falling back to p when n and z are not set.
-//      // That however would require other code to be re factored.
-//      String[] formatBits = formatStrIn.split(";");
-//      int i = cellValue > 0.0 ? 0 : cellValue < 0.0 ? 1 : 2; 
-//      String formatStr = (i < formatBits.length) ? formatBits[i] : formatBits[0];
+        Locale originalUserLocale = LocaleUtil.getUserLocale();
+        if (originalUserLocale != locale) {
+            LocaleUtil.setUserLocale(locale);
```

Comment:
Unfortunately this will create very hard to debug problems in multi-threaded environments, because it changes the current locale for all threads and if one is working on another document with another locale it will break in very strange ways. One option would be locking, however this will be bad for overall performance as it would need to lock at every place that is using LocaleUtils, let me take a look if we can do it differently here somehow...